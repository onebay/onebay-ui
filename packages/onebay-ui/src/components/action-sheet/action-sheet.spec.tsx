import { mount } from '@vue/test-utils'
import ActionSheet from './index'
import ActionSheetItem from './components/item'
import ActionSheetHeader from './components/header'
import ActionSheetFooter from './components/footer'

const factory = (
  values = {},
  slots = {
    default: [<ActionSheetItem>Button A</ActionSheetItem>, <ActionSheetItem>Button B</ActionSheetItem>]
  }
) => {
  return mount(ActionSheet, {
    slots,
    components: {
      ActionSheetHeader,
      ActionSheetFooter,
      ActionSheetItem
    },
    props: { ...values }
  })
}

describe('ActionSheet Snap', () => {
  it('render initial ActionSheet', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render opened ActionSheet', () => {
    const wrapper = factory({
      isOpened: true
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render opened ActionSheet -- props cancelText', () => {
    const wrapper = factory({
      isOpened: true,
      cancelText: 'Cancel'
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render opened ActionSheet -- props title', () => {
    const wrapper = factory({
      isOpened: true,
      title: 'After clearing the location information, others will not be able to see you\r\nLine breaks can be escaped'
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render opened ActionSheet -- props completed', () => {
    const wrapper = factory({
      isOpened: true,
      cancelText: 'Cancel',
      title: 'After clearing the location information, others will not be able to see you\r\nLine breaks can be escaped'
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('Change ActionSheet -- props isOpened', async () => {
    const wrapper = factory({
      cancelText: 'Cancel',
      title: 'After clearing the location information, others will not be able to see you\r\nLine breaks can be escaped'
    })
    await wrapper.setProps({ isOpened: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('Change ActionSheet -- props isOpened', async () => {
    const wrapper = factory({
      isOpened: true,
      cancelText: 'Cancel',
      title: 'After clearing the location information, others will not be able to see you\r\nLine breaks can be escaped'
    })
    await wrapper.setProps({ isOpened: false })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('ActionSheet Behavior ', () => {
  beforeEach(() => { })
  it('ActionSheet onCancel & onClose', () => {
    const wrapper = factory(
      {
        isOpened: true,
        cancelText: 'Cancel',
        title: 'After clearing the location information, others will not be able to see you\r\nLine breaks can be escaped'
      },
      {
        default: [<ActionSheetItem>Button A</ActionSheetItem>]
      }
    )
    wrapper.find('.ob-action-sheet__footer').trigger('click')
    expect(wrapper.emitted()).toHaveProperty('cancel')

    wrapper.find('.ob-action-sheet__overlay').trigger('click')
    expect(wrapper.emitted()).toHaveProperty('close')
  })

  it('ActionSheet click event on ActionSheetItem', () => {
    const wrapper = mount(
      <ActionSheet>
        <ActionSheetItem>Button A</ActionSheetItem>
      </ActionSheet>
    )

    expect(wrapper.element).toMatchSnapshot()
    wrapper.trigger('touchmove')
    wrapper.find('.ob-action-sheet__item').trigger('click')

    const comp = wrapper.findComponent({ name: 'ActionSheetItem' })
    expect(comp.emitted()).toHaveProperty('click')
  })
})
