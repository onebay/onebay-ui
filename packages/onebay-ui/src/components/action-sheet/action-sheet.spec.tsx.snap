// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionSheet Behavior  ActionSheet click event on ActionSheetItem 1`] = `
<div
  class="ob-action-sheet"
>
  <div
    class="ob-action-sheet__overlay"
  />
  <div
    class="ob-action-sheet__container"
  >
    
    <div
      class="ob-action-sheet__body"
    >
      
      
      <div
        class="ob-action-sheet__item"
      >
        
        Button A
        
      </div>
      
      
    </div>
    <div
      class="ob-action-sheet__footer"
    >
      
      
      
    </div>
  </div>
</div>
`;

exports[`ActionSheet Snap Change ActionSheet -- props isOpened 1`] = `
<div
  class="ob-action-sheet ob-action-sheet--active"
>
  <div
    class="ob-action-sheet__overlay"
  />
  <div
    class="ob-action-sheet__container"
  >
    <div
      class="ob-action-sheet__header"
    >
      
      After clearing the location information, others will not be able to see you
Line breaks can be escaped
      
    </div>
    <div
      class="ob-action-sheet__body"
    >
      
      
      <div
        class="ob-action-sheet__item"
      >
        
        Button A
        
      </div>
      <div
        class="ob-action-sheet__item"
      >
        
        Button B
        
      </div>
      
      
    </div>
    <div
      class="ob-action-sheet__footer"
    >
      
      Cancel
      
    </div>
  </div>
</div>
`;

exports[`ActionSheet Snap Change ActionSheet -- props isOpened 2`] = `
<div
  class="ob-action-sheet"
>
  <div
    class="ob-action-sheet__overlay"
  />
  <div
    class="ob-action-sheet__container"
  >
    <div
      class="ob-action-sheet__header"
    >
      
      After clearing the location information, others will not be able to see you
Line breaks can be escaped
      
    </div>
    <div
      class="ob-action-sheet__body"
    >
      
      
      <div
        class="ob-action-sheet__item"
      >
        
        Button A
        
      </div>
      <div
        class="ob-action-sheet__item"
      >
        
        Button B
        
      </div>
      
      
    </div>
    <div
      class="ob-action-sheet__footer"
    >
      
      Cancel
      
    </div>
  </div>
</div>
`;

exports[`ActionSheet Snap render initial ActionSheet 1`] = `
<div
  class="ob-action-sheet"
>
  <div
    class="ob-action-sheet__overlay"
  />
  <div
    class="ob-action-sheet__container"
  >
    
    <div
      class="ob-action-sheet__body"
    >
      
      
      <div
        class="ob-action-sheet__item"
      >
        
        Button A
        
      </div>
      <div
        class="ob-action-sheet__item"
      >
        
        Button B
        
      </div>
      
      
    </div>
    <div
      class="ob-action-sheet__footer"
    >
      
      
      
    </div>
  </div>
</div>
`;

exports[`ActionSheet Snap render opened ActionSheet -- props cancelText 1`] = `
<div
  class="ob-action-sheet ob-action-sheet--active"
>
  <div
    class="ob-action-sheet__overlay"
  />
  <div
    class="ob-action-sheet__container"
  >
    
    <div
      class="ob-action-sheet__body"
    >
      
      
      <div
        class="ob-action-sheet__item"
      >
        
        Button A
        
      </div>
      <div
        class="ob-action-sheet__item"
      >
        
        Button B
        
      </div>
      
      
    </div>
    <div
      class="ob-action-sheet__footer"
    >
      
      Cancel
      
    </div>
  </div>
</div>
`;

exports[`ActionSheet Snap render opened ActionSheet -- props completed 1`] = `
<div
  class="ob-action-sheet ob-action-sheet--active"
>
  <div
    class="ob-action-sheet__overlay"
  />
  <div
    class="ob-action-sheet__container"
  >
    <div
      class="ob-action-sheet__header"
    >
      
      After clearing the location information, others will not be able to see you
Line breaks can be escaped
      
    </div>
    <div
      class="ob-action-sheet__body"
    >
      
      
      <div
        class="ob-action-sheet__item"
      >
        
        Button A
        
      </div>
      <div
        class="ob-action-sheet__item"
      >
        
        Button B
        
      </div>
      
      
    </div>
    <div
      class="ob-action-sheet__footer"
    >
      
      Cancel
      
    </div>
  </div>
</div>
`;

exports[`ActionSheet Snap render opened ActionSheet -- props title 1`] = `
<div
  class="ob-action-sheet ob-action-sheet--active"
>
  <div
    class="ob-action-sheet__overlay"
  />
  <div
    class="ob-action-sheet__container"
  >
    <div
      class="ob-action-sheet__header"
    >
      
      After clearing the location information, others will not be able to see you
Line breaks can be escaped
      
    </div>
    <div
      class="ob-action-sheet__body"
    >
      
      
      <div
        class="ob-action-sheet__item"
      >
        
        Button A
        
      </div>
      <div
        class="ob-action-sheet__item"
      >
        
        Button B
        
      </div>
      
      
    </div>
    <div
      class="ob-action-sheet__footer"
    >
      
      
      
    </div>
  </div>
</div>
`;

exports[`ActionSheet Snap render opened ActionSheet 1`] = `
<div
  class="ob-action-sheet ob-action-sheet--active"
>
  <div
    class="ob-action-sheet__overlay"
  />
  <div
    class="ob-action-sheet__container"
  >
    
    <div
      class="ob-action-sheet__body"
    >
      
      
      <div
        class="ob-action-sheet__item"
      >
        
        Button A
        
      </div>
      <div
        class="ob-action-sheet__item"
      >
        
        Button B
        
      </div>
      
      
    </div>
    <div
      class="ob-action-sheet__footer"
    >
      
      
      
    </div>
  </div>
</div>
`;
